variables.tf

```diff
@@ -240,6 +234,33 @@ variable "map_users" {
   default     = []
 }
 
+variable "ingress_nginx_config" {
+  type = object({
+    enable          = bool
+    helm_repository = string
+    helm_version    = string
+    chart_values    = map(any)
+  })
+  description = "Input configuration for ingress-nginx service deployed with helm release. By setting key 'enabled' to 'true', ingress-nginx service will be deployed. 'helm_repository' is an URL for the repository of ingress-nginx helm chart, where 'helm_version' is its respective version of a chart. 'chart_values' is used for changing default values.yaml of an ingress-nginx chart."
+  default = {
+    enable          = false
+    helm_repository = "https://kubernetes.github.io/ingress-nginx"
+    helm_version    = "4.1.4"
+    chart_values = {
```

Comment:
It shall be possible to **only add additional** `chart_values`, without knowing the other defaults, when using the reference architecture as a submodule, as it is the case in SIMPHERAs infrastructure repo.

I am unsure, whether this is possible or whether always the complete default value needs to be specified.

Reply:
It is possible to leave chart_values as a empty dictionary. If this is set to empty dictionary, created load balancer will be internal, and not internet-facing.
Though, even this we can move to the "modules\eks_addons\templates\nginx_values.yaml", that way it would be set as default. But I think we should leave it here for user to how he can override default values.yaml.

Reply:
We had a discussion and agreed to seperate variables.
One to enable/disable ingress_nginx creation, and another one for configuring it. That way user only have to set one variable to true or false, changing/adding chart values is only optional.

Default values for ingress_nginx are defined and "hidden" from in `modules/k8s-eks-addons/templates/nginx_values.yaml`. This way `chart_values` are only used to add additional values, e.g. to set load balancer as internet facing or internal.

Reply:
If we would update the terraform version to 1.3, we could use [Optional Object Type Attributes](https://developer.hashicorp.com/terraform/language/v1.3.x/expressions/type-constraints#optional-object-type-attributes) to allow the user to set only some values like `helm_version`.

Reply:
Terraform version updated and optional object type implemented. Please check.